### DISCUSSION TITLE
Searching JSON db

### DISCUSSION THREAD

**Discussion by pbhuter** [msiemens/tinydb#492] [Q&A]
Hello:
I have a JSON db with multiple levels, as shown in the attached JSON schema. Would anyone be able to help me get started structuring a query to search this db? I would like to search each set of "Points" to compare against another set of points and return the "Sc" value for the closest match.

[schema.txt](https://github.com/msiemens/tinydb/files/9985021/schema.txt)

**Comment by msiemens** [comment]
Hey @pbhuter,

TinyDB has [`all` and `any` operators](https://tinydb.readthedocs.io/en/latest/usage.html#advanced-queries) that return documents where either all or at least one item of a list matches a query. However, the query system in TinyDB isn't really designed to compare documents in the database against each other (there's for example no built-in JOIN), nor is finding the document with the largest/smallest value of some field supported. If this is what your use case requires, I'm afraid it might stretch what TinyDB was built for.

Cheers!

**Reply by pbhuter** [reply]
Hey @msiemens 

This is the same project as your other reply (https://github.com/msiemens/tinydb/discussions/493#discussioncomment-4185070), which was the idea I came up with to address what you are talking about. I am going to give that a go, and I will post on that thread the results.

**Reply by msiemens** [reply]
Sounds good, I'm curious about the results 🙂

**Reply by pbhuter** [reply]
Thank you. I will let you know.
